ui/src/app/utils/access-utils.tsx

```diff
@@ -361,6 +361,45 @@ export const useIsUserDisabled = () => {
   return disabled;
 };
 
+export const useNeedsToAcceptTOS = () => {
+  const { authLoaded, isSignedIn } = useStore(authStore);
+  const [userRequiredToAcceptTOS, setUserRequiredToAcceptTOS] = useState<
+    boolean | undefined
```

Comment:
do we need to use 3-way logic here?  We're asking a binary question: do we need to interrupt the user by throwing up a TOS screen?  (also maybe a good name for this would be `useRequireTOS()`)

if (!auth || !signIn) -> false

otherwise return the result of the API call

Reply:
Done

Reply:
please make this a boolean only